app/assets/javascripts/app/views/stream_post_views.js

```diff
@@ -28,9 +28,12 @@ app.views.StreamPost = app.views.Post.extend({
                      ".permalink"].join(", "),
 
   initialize : function(){
-    var personId = this.model.get("author").id;
-    app.events.on("person:block:"+personId, this.remove, this);
-
+    var baseUrl = Backbone.history.fragment.split("/")[0];
+      // If we are on a user page, we don't want to remove posts on block
+    if (baseUrl !== "people" && baseUrl !== "u") {
```

Comment:
Maybe you can check the `page-people` class on the `<body>`? That's the only way I know to detect on which page you are, but I don't really know the frontend, so maybe @svbergerem still has a better idea? ;)